### DISCUSSION TITLE
How about Scala support?

### DISCUSSION THREAD

**Discussion by linux-china** [jbangdev/jbang#1167] [Ideas]
Kotlin,  Groovy  and Scala are most 3 popular JVM languages.  Some developers choose [Scala CLI](https://scala-cli.virtuslab.org/), but the model is different with JBang.  

```scala
// using lib com.lihaoyi::os-lib:0.7.8

def filesByExtension(
  extension: String, 
  dir: os.Path = os.pwd): Seq[os.Path] = 
    os.walk(dir).filter { f =>
      f.last.endsWith(s".$extension") && os.isFile(f)
    }
```

Code snippet from [ammonite](https://ammonite.io/) 

```scala
import $ivy.`com.lihaoyi::scalatags:0.7.0 compat`, scalatags.Text.all._

val rendered = div("Moo").render

```

**Comment by maxandersen** [comment]
Scala has Multiple cli efforts. Not sure which makes sense to enable.

**Reply by linux-china** [reply]
Yes,   some Chinese developers asked me this question.  They think `//DEPS ` is nice design, and it could be used for Scala too.  :)

**Reply by maxandersen** [reply]
There is s jbang like cli for scala - the name escapes me though.
